python/whylogs/experimental/api/constraints/__init__.py

```diff
@@ -0,0 +1,171 @@
+from logging import getLogger
+from typing import List, Optional, Union
+
+import yaml
+from typing_extensions import TypedDict
+
+from whylogs.core.constraints import PrefixCondition
+from whylogs.core.constraints.factories import (
+#    IsInRangeParams,
+#    NoMissingValues,
+#    ColumnIsProbablyUniqueParams,
+    column_is_probably_unique,
+    condition_meets,
+#    ConditionMeetsParams,
+    count_below_number,
+#    CountBelowNumber,
+    distinct_number_in_range,
+#    DistinctNumberInRangeParams,
+    is_in_range,
+    is_non_negative,
+#    IsNonNegativeParams,
+    no_missing_values,
+)
+from whylogs.core.constraints.metric_constraints import (
+    Constraints,
+    DatasetConstraint,
+    MetricConstraint,
+)
+
+logger = getLogger(__name__)
+
+"""
+constraints_mapping = {
+    "no_missing_values": {
+        "constraint_function": no_missing_values,
+        "whylabs_datatypes": ["string", "integral", "fractional", "bool", "unknown"],
+        "parameters": NoMissingValues,
+    },
+    "is_in_range": {
+        "constraint_function": is_in_range,
+        "whylabs_datatypes": ["integral", "fractional"],
+        "parameters": IsInRangeParams,
+    },
+    "column_is_probably_unique": {
+        "constraint_function": column_is_probably_unique,
+        "whylabs_datatypes": ["string", "integral"],
+        "parameters": ColumnIsProbablyUniqueParams,
+    },
+    "distinct_number_in_range": {
+        "constraint_function": distinct_number_in_range,
+        "whylabs_datatypes": ["string", "integral"],
+        "parameters": DistinctNumberInRangeParams,
+    },
+    "count_below_number": {
+        "constraint_function": count_below_number,
+        "whylabs_datatypes": ["string", "integral", "fractional", "bool", "unknown"],
+        "parameters": CountBelowNumber,
+    },
+    "is_non_negative": {
+        "constraint_function": is_non_negative,
+        "whylabs_datatypes": ["integral", "fractional"],
+        "parameters": IsNonNegativeParams,
+    },
+    "condition_meets": {
+        "constraint_function": condition_meets,
+        "whylabs_datatypes": ["string", "integral", "fractional", "bool", "unknown"],
+        "parameters": ConditionMeetsParams,
+    },
+}
+"""
+
+def assemble_constraint(constraint_dict: dict) -> Optional[TypedDict]:
+    constraint_name = None
+    if constraint_dict.get("expression") and constraint_dict.get("name"):
+        condition = PrefixCondition(constraint_dict.get("expression"))
+        returned_constraint = DatasetConstraint(condition=condition, name=constraint_dict.get("name"))
+        return returned_constraint
+    """
```

Comment:
why is this code block sitting here?

Reply:
Also not yet fully moved over from the hackathon branch